### ISSUE TITLE
How do we futureproof item fetch quest difficulty against increased item variety?

### ISSUE THREAD

**Issue by MrHrulgin** [CleverRaven/Cataclysm-DDA#71629]
### Is your feature request related to a problem? Please describe.

Any quest that needs a number of a specific item is (presumably) tuned with an expectation that acquiring those items is going to be a particular level of difficulty. If I need to bring back 25 blankets, that quest is designed with an expectation that it's part of an item pool that has a certain number of items in it. But if one day someone gets a bee in their bonnet about bedding and expands that pool to have duvets and sheets and fitted sheets and shams and coverlets and decorative pillows without being very careful about spawning numbers and item weights, the amount of time it'll take to find the needed blankets will rise. Any quest where a specific item is needed will be vulnerable to this over time.

### Solution you would like.

For quests to have more flexible objectives, except when there's a specified spot to acquire the object. My understanding is that it's not currently possible in the quest logic, but being able to have quests accept anything in a given pool of items would make the targeted quest difficulty more resilient in the face of other contributions to the game. If someone who needs antibiotics could need X strength of antibiotics rather than needing a given item. If any items from a "bedding" group would count rather than needing blankets. If anything with a wood sawing quality of 2 or higher would work, rather than needing a circular saw. 

### Describe alternatives you have considered.

Needing to review quest balance over and over again as the game shifts and more things are added.

**Comment by IdleSol** [comment]
Isn't that what you're saying?
```
    "id": "MISSION_LUMBERMILL_GET_6_BLANKETS",
    "type": "mission_definition",
    "name": "Find 6 Blankets",
    "description": "Obtain 6 blankets or quilts in exchange for credit with the loggers.",
    "goal": "MGOAL_CONDITION",
    "difficulty": 5,
    "value": 4000,
    "//": "Dynamic response dialogs would be more natural to accept so many alternate items here.",
    "goal_condition": {
      "or": [
        { "u_has_items": { "item": "blanket", "count": 6 } },
        { "u_has_items": { "item": "fur_blanket", "count": 6 } },
        { "u_has_items": { "item": "electric_blanket", "count": 6 } },
        { "u_has_items": { "item": "quilt", "count": 6 } },
        { "u_has_items": { "item": "quilt_patchwork", "count": 6 } }
      ]
    },
```

**Comment by MrHrulgin** [comment]
Yes, sort of. But that's still a manual list of items that wouldn't automatically handle an expansion of the kinds of bedding available, and only completes if you have 6 of one of the things on the list. A blanket, two quilts and 3 patchwork quilts would not be sufficient.

**Comment by PatrikLundell** [comment]
Well, technically you COULD use @IdleSol's example by having a count of 1 for each entry and repeat that entry 25 times (or 6 in that example) with "and" conditions. However, that would be very unwieldy and would remain unwieldy even if each group was reduced to a single entry (like how you can define "LIST" in crafting recipes).

**Comment by Inglonias** [comment]
Item variants seems like a great way to do this. For instance, with the exception of the shark plushie, which is a reference to a specific real-world object, pretty much every plushie in the game has the same item ID.

**Comment by PatrikLundell** [comment]
No, item variant aren't applicable, because they are cosmetic variants of a base item, such as differing pattern on T-shirts, but made from the same types and amount of materials. Thus, the plushie is a thingie made out of a fixed set of standard amounts of materials but shaped differently. This can work because they're deconstructed into standard patches, rather than shaped bits of fabric (plus standard amounts of standard stuffing stuff, I assume). If you wanted to be able to stuff them with say hay or fabric scraps rather than standard stuffing it wouldn't work.

Blankets, on the other hand, are different items serving the same purpose, but constructed out of different materials and, crucially, deconstructed into different materials. In fact, a fair bit of effort has been spent on splitting same purpose items into multiple distinct items in order to ensure they deconstruct back into the materials they were made of.
That could also have been done by storing construction material info in the items themselves, in a fashion akin to how the "same" food dish differs in nutrients based on the input (but in that case it's simplified to just nutrients, not actual inputs).

**Comment by NthTech** [comment]
It's not mentioned in the missions_json documentation, but there's apparently code for handling missions that require item groups rather than specific items. https://github.com/CleverRaven/Cataclysm-DDA/blob/00108ad58c371d7dd436391516816b6eb7e4472e/src/mission.cpp#L383
Is it possible to use that to solve the bedding example problem?

**Comment by Cenness** [comment]
>Is it possible to use that to solve the bedding example problem?

Yes, when that option will be made to work. See previous discussion at https://github.com/CleverRaven/Cataclysm-DDA/issues/71215

**Comment by MrHrulgin** [comment]
I submitted this one two weeks later, as I ran into more instances of the same broader issue. When I submitted the question of blankets, it seemed a more narrow issue, but with further investigation it seemed to be a broader question of balancing, and so I opened a separate issue to talk about this broader question. If the word of Kevin is that the difficulty will vary and everyone is just kind of pulling a number that they think is reasonable out of thin air, how do we even have a conversation about quest balance? If you're going to axe one of the two posts, I'd really prefer to hold onto this broader discussion.
